src/rgw/rgw_acl.h

```diff
@@ -214,20 +217,25 @@ struct ACLReferer {
       perm(perm) {
   }
 
-  bool is_match(std::string http_referer) const {
-    if (http_referer == url_spec) {
+  bool is_match(boost::string_ref http_referer) const {
+    const auto http_host = get_http_host(http_referer);
+    if (!http_host) {
+      return false;
+    }
+
+    boost::string_ref host_spec(url_spec);
+    if (*http_host == host_spec) {
       return true;
     }
 
-    if (http_referer.length() < url_spec.length()) {
+    if (http_host->length() < host_spec.length()) {
       return false;
     }
 
-    if ('.' == url_spec[0]) {
+    if ('.' == host_spec[0]) {
       /* Wildcard support: a referer matches the spec when its last char are
        * perfectly equal to spec. */
-      return !http_referer.compare(http_referer.length() - url_spec.length(),
-                                   url_spec.length(), url_spec);
+      return (*http_host).ends_with(host_spec);
```

Comment:
The `->` operator can be applied here as well. I've missed that last time, sorry.